File: 0045-jump-game-ii/0045-jump-game-ii.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var jump = function(nums) {
  let [jumps, end, farCanReach] = [0, 0, 0];
  
  for (let i = 0; i < nums.length - 1; i++) {
    farCanReach = Math.max(farCanReach, nums[i] + i);
    if (i === end) {
      end = farCanReach;
      jumps++;
    }
  }
  
  return jumps;
};

File: 0045-jump-game-ii/0045-jump-game-ii.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        jumps = end = can_farthest = 0
        for i in range(len(nums) - 1):
            can_farthest = max(can_farthest, nums[i] + i)
            if i == end:
                end = can_farthest
                jumps += 1
        return jumps